CMakeLists.txt

```diff
@@ -40,7 +40,7 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_C_COMPILER $ENV{CC})
 set(CMAKE_CXX_COMPILER $ENV{CXX})
 
-if(CMAKE_COMPILER_IS_GNUCXX)
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER MATCHES ".*clang")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wno-error=deprecated-declarations")
 
     option(CMAKE_CXX11_ABI "Enable the GLIBCXX11 ABI" ON)
```

Comment:
Does this work with Clang?

Reply:
Yes. This is the only part of the change that's required for clang to work. The rest is just for consistency with gcc, which clang also supports. (clang does support using libstdc++ as a C++ standard library) 

I couldn't manage to get clang to work without the C++11 ABI. I'm not sure if I'm missing something there, or whether clang supports the C++03 ABI at all. It supports C++03, but I haven't used a C++03 libstdc++ in a long time.

Reply:
I only managed to get clang to work on my environment with the C++11 ABI; using the C++03 fails to link due to linking errors (undefined references, caused by C++11 abis being referenced but not available).

Not sure if it is possible to get clang to work with the C++03 ABI. I imagine it is possible, but I don't need it, so didn't pursue it.

Shall I error here if the compiler is clang and this option is off?

Reply:
I've opened and issue, and opened a PR with the check: https://github.com/rapidsai/rmm/pull/549